homeassistant/components/knx/config_flow.py

```diff
@@ -614,12 +638,3 @@ async def async_step_communication_settings(
             data_schema=vol.Schema(data_schema),
             last_step=True,
         )
-
-
-async def scan_for_gateways(stop_on_found: int = 0) -> list[GatewayDescriptor]:
-    """Scan for gateways within the network."""
-    xknx = XKNX()
-    gatewayscanner = GatewayScanner(
-        xknx, stop_on_found=stop_on_found, timeout_in_seconds=2
-    )
-    return await gatewayscanner.scan()
```

Comment:
You could have made this function a async generator instead, and handled the no timeout iteration inside here. Then the config flow would have been simpler.

Reply:
I'll look into it. Not done with the flow anyway 😃
For now I'll fix conflicts of #82724 and try to implement another small feature before beta 😬